packages/mds-schema-validators/validators.ts

```diff
@@ -418,4 +455,67 @@ export function rawValidatePolicy(policy: Policy): Joi.ValidationResult<Policy>
   return Joi.validate(policy, policySchema)
 }
 
+export function validateDevice(device: Device) {
+  const { error } = Joi.validate(device, deviceSchema)
+  return error
+}
+
+const validateTripEvent = (event: VehicleEvent) => {
+  const { error } = Joi.validate(event, tripEventSchema)
+  return error
+}
+
+const validateProviderPickUpEvent = (event: VehicleEvent) => {
+  const { error } = Joi.validate(event, providerPickUpEventSchema)
+  return error
+}
+
+const validateServiceEndEvent = (event: VehicleEvent) => {
+  const { error } = Joi.validate(event, serviceEndEventSchema)
+  return error
+}
+
+const validateDeregisterEvent = (event: VehicleEvent) => {
+  const { error } = Joi.validate(event, deregisterEventSchema)
+  return error
+}
+
+export const validateEvent = (event: VehicleEvent) => {
+  const { event_type } = event
+
+  const TRIP_EVENTS: string[] = [
+    VEHICLE_EVENTS.trip_start,
+    VEHICLE_EVENTS.trip_end,
+    VEHICLE_EVENTS.trip_enter,
+    VEHICLE_EVENTS.trip_leave
+  ]
+
+  if (TRIP_EVENTS.includes(event_type)) {
+    return validateTripEvent(event)
+  }
+  if (event_type === VEHICLE_EVENTS.provider_pick_up) {
+    return validateProviderPickUpEvent(event)
+  }
+  if (event_type === VEHICLE_EVENTS.service_end) {
+    return validateServiceEndEvent(event)
+  }
+  if (event_type === VEHICLE_EVENTS.deregister) {
+    return validateDeregisterEvent(event)
+  }
```

Comment:
Should we verify that `event_type_reason` is `null`, `undefined`, or _empty_ for the other event types?

Reply:
That is handled in the `eventSchema` because we don't validate with `allowUnknown` in the fall-through case.